src/protections/NsfwProtection.ts

```diff
@@ -47,23 +47,50 @@ export class NsfwProtection extends Protection {
         if (event['type'] === 'm.room.message') {
             const content = event['content'] || {};
             const msgtype = content['msgtype'] || 'm.text';
-            const isMedia = msgtype === 'm.image';
+            const formattedBody = content['formatted_body'] || '';
+            const isMedia = msgtype === 'm.image' || formattedBody.toLowerCase().includes('<img');
 
             if (isMedia) {
-                const mxc = content["url"];
+                let mxc;
+                if (formattedBody) {
+                    const body = content["body"]
+                    const mxcRegex = /(mxc?:\/\/[^\s]+)/g;
+                    const url = body.match(mxcRegex)
+                    if (url) {
+                        mxc = url[0]
```

Comment:
Another option is to `JSON.stringify(content)` then extract all the MXC URIs from that, check them individually, then redact as needed. This would also make the check slightly more resilient to weird event types too.

Rough code would be:

```typescript
public async handleEvent(mjolnir: Mjolnir, roomId: string, event: any): Promise<any> {
    const content = JSON.stringify(event['content']);
    if (!content.toLowerCase().includes("mxc")) {
        return; // not something we need to check
    }
    const mxcs = content.match(/(mxc?:\/\/[^\s]+)/gim); // TODO: Double check flags for multiline, multiple matches, and case insensitivity
    for (const mxc in mxcs) {
        if (isNsfw(mxc)) {
            // TODO: Redact and notify management room
            break;
        }
    }
}
```